tests/platform_tests/counterpoll/test_counterpoll_watermark.py

```diff
@@ -0,0 +1,233 @@
+    verify_counterpoll_status(duthost, [tested_counterpoll], ENABLE)
+
+    # verify QUEUE or PG maps are generated into COUNTERS_DB after enabling relevant counterpoll
+    with allure.step("Verifying MAPS in COUNTERS_DB on {}...".format(duthost.hostname)):
+        maps_dict = RELEVANT_MAPS[tested_counterpoll]
+        maps_to_verify = maps_dict[MAPS]
+        for map_to_verify in maps_to_verify:
+            map_prefix = map_to_verify['prefix']
+            maps = map_to_verify[MAPS]
+            map_output = redis_get_keys(duthost, 'COUNTERS_DB', MAPS_LONG_PREFIX.format(map_prefix))
+            map = []
+            failed = ""
+            for map_entry in maps:
+                map.append(map_entry)
+            msg = "no {} maps found in COUNTERS_DB".format(map_prefix)
+            pytest_assert(map_output, msg)
+            for line in map_output:
+                try:
+                    map.remove(line)
+                except ValueError:
+                    failed = "MAP {} was not found in {} MAPS list".format(line, map_prefix)
+            pytest_assert("" == failed, failed)
+            pytest_assert(len(map) == 0, "{} maps mismatch, one or more queue was not found in redis COUNTERS_DB"
+                          .format(map_prefix))
+
+    failed_list = []
+    with allure.step("Verifying {} STATS in FLEX_COUNTER_DB on {}...".format(tested_counterpoll, duthost.hostname)):
+        stats_output = redis_get_keys(duthost, 'FLEX_COUNTER_DB', '*{}*'.format(map_prefix))
+        counted = 0
+        # build expected counterpoll stats vs unexpected
+        expected_types = []
+        unexpected_types = []
+        for counterpoll, v in RELEVANT_MAPS.items():
+            types_to_check = v[CounterpollConstants.TYPE]
+            if counterpoll in tested_counterpoll:
+                for type in types_to_check:
+                    expected_types.append(FLEX_COUNTER_PREFIX + type)
+            else:
+                for type in types_to_check:
+                    unexpected_types.append(FLEX_COUNTER_PREFIX + type)
+        logging.info("expected types for for counterpoll {}:\n{}".format(tested_counterpoll, expected_types))
+        logging.info("unexpected types for for counterpoll {}:\n{}".format(tested_counterpoll, unexpected_types))
+        for line in stats_output:
+            for expected in expected_types:
+                if expected in line:
+                    counted += 1
+            for unexpected in unexpected_types:
+                if unexpected in line:
+                    failed_list.append("found for {} unexpected stat counter in FLEX_COUNTER_DB: {}"
+                                       .format(tested_counterpoll, line))
+        logging.info("counted {} {} STATs type in FLEX_COUNTER_DB on {}..."
+                     .format(counted, tested_counterpoll, duthost.hostname))
+        failed_list
+        pytest_assert(counted > 0, "counted {} for {}".format(counted, tested_counterpoll))
+
```

Comment:
Do we need assert `unexpected == 0` as well?

Reply:
it should have been handled by the failed_list instead, will fix it and update the PR

Reply:
Added assertion for failed_list, it's accumulating the unexpected if there are any